frame/treasury/src/lib.rs

```diff
@@ -534,6 +665,68 @@ impl<T: Config<I>, I: 'static> Pallet<T, I> {
 	/// Spend some money! returns number of approvals before spend.
 	pub fn spend_funds() -> Weight {
 		let mut total_weight = Weight::zero();
+		let mut total_spent = BalanceOf::<T, I>::zero();
+		let mut missed_proposals: u32 = 0;
+		let proposals_len = PendingPayments::<T, I>::count();
+
+		Self::deposit_event(Event::ProcessingProposals { waiting_proposals: proposals_len });
+
+		for key in PendingPayments::<T, I>::iter_keys() {
+			if let Some(mut p) = PendingPayments::<T, I>::get(key) {
+				match p.payment_id {
+					None =>
+						if let Ok(id) = T::Paymaster::pay(&p.beneficiary, p.asset_id, p.value) {
+							total_spent += p.normalized_value;
+							p.payment_id = Some(id);
+							PendingPayments::<T, I>::set(key, Some(p));
+						} else {
+							missed_proposals = missed_proposals.saturating_add(1);
+						},
+					Some(payment_id) => match T::Paymaster::check_payment(payment_id) {
+						PaymentStatus::Failure => {
+							// try again in the next `T::SpendPeriod`.
+							missed_proposals = missed_proposals.saturating_add(1);
+							Self::deposit_event(Event::ProposalPaymentFailure {
+								proposal_index: key,
+								asset_id: p.asset_id,
+								amount: p.value,
+							});
+							// Force the payment to none, so a fresh payment is sent during the next
+							// T::SpendPeriod.
+							p.payment_id = None;
+							PendingPayments::<T, I>::set(key, Some(p));
+						},
+						PaymentStatus::Success => {
+							PendingPayments::<T, I>::remove(key);
+							Self::deposit_event(Event::ProposalPaymentSuccess {
+								proposal_index: key,
+								asset_id: p.asset_id,
+								amount: p.value,
+							});
+						},
+						// PaymentStatus::InProgress and PaymentStatus::Unknown indicate that the
```

Comment:
may be worth emitting event, so we know it was processed

Reply:
I've added an event for when the payment is created